38.md

```diff
@@ -0,0 +1,170 @@
+
+If new members are added, then its the responsibility of the creator's client to share the shared-secret with them as mentioned in kind 140 section. 
+
+## Kind 142: Send encrypted message to encrypted group channel
+
+
+Clients SHOULD use [NIP-10](10.md) marked "e" tags to recommend a relay and specify whether it is a reply or root message.
+
+Clients SHOULD append [NIP-10](10.md) "p" tags to replies.
+
+### Format
+
+Root message:
+
+```json
+{
+    "content": <string>,
+    "tags": [["e", <kind_140_event_id>],
+    ...
+}
+```
+
+Reply to another message:
+
+```json
+{
+    "content": <string>,
+    "tags": [
+        ["e", <kind_140_event_id>],
+        ["e", <kind_142_event_id_replied_to>], 
+        ["p", <pubkey> <relay-url>],
+        
+        ...
+    ],
+    ...
+}
+```
+
+## Kind 104: Communicate the Shared-Secret
+
+Upon adding a new member to a encrypted group of kind 140, the creator of the channel should send a direct message ( of kind 104) to each of the other participants, which will have the channel's shared-secret private/public keys. The direct message should of `kind` 104; it must follow the message construction rules as are there for `kind 4` messages, except for the difference in value of `kind`.
+
+The format of that message is:
+
+`App Encrypted Channels: inviting you to encrypted channel <channel id> encrypted using private public keys <shared private key> <corresponding shared public key>`
+
+The three id's mentioned should each be 64 byte long hex encoded strings. No quotes are used.
+
+Clients that support this NIP will read that and know that they're invited to join the given encrypted channel, and they can use the given secret key to send/read messages in that channel.
+
+Other clients will decrypt this message, and ignore it, but the user can read that they have been invited to such a channel.
+
+### Encryption
+
+All participants should ideally have the shared secret sent to them by the creator of the channel. If such a message is present, the client should read the shared secret from that message, and then encrypt any 142 kind message they want to send using it.
+
+Only the given shared-secret private and public keys are used to encrypt or decrypt the messages. 
+
+## Adding and Removing Participants 
+
+### Adding Participants In Same Group
+
+To add participants to the group, the admin or creator of the group emits a kind 141 message with the updated p tags, which also include the pubkeys of new members. Along with this, the admin shall also communicate to the newly added members the group shared-secret, using the same mechanism as mentioned in section for kind 104 above. 
+
+### Removing Participants
```

Comment:
What if a user wants to remove themself? Obviously the client can just ignore messages, but it seems like a good client should also notify the relay that they left the room, and the admin should remove them from the list and roll the keys.

Reply:
that's a good idea